acceptance/tests/language/sensitive_data_type.rb

```diff
@@ -0,0 +1,99 @@
+       :assertions => {:assert_match => "8:sekrit8"}},
+    {:type => 'notify', :parameters => {:namevar => "9:meh", :message => '"9:${unwrap(Sensitive.new(\'sekrit9\'))}"'},
+       :assertions => {:assert_match => "9:sekrit9"}},
+    {:type => 'notify', :parameters => {:namevar => "A:meh", :message => '"A:${unwrap($mehA)}"'}, :pre_code => '$mehA=Sensitive.new("sekritA")',
+       :assertions => {:assert_match => "A:sekritA"}},
+    {:type => 'notify', :parameters => {:namevar => "B:meh", :message => '"B:${$mehB.unwrap}"'}, :pre_code => '$mehB=Sensitive.new("sekritB")',
+       :assertions => {:assert_match => "B:sekritB"}},
+    {:type => 'notify', :parameters => {:namevar => "C:meh", :message => '"C:${$mehC.unwrap |$unwrapped| { "blk_${unwrapped}_blk" } } nonblk_${mehC}_nonblk"'}, :pre_code => '$mehC=Sensitive.new("sekritC")',
+       :assertions => {:assert_match => ["C:blk_sekritC_blk", "nonblk_#{notify_redacted}_nonblk"]}},
+    # for --show_diff
+    {:type => 'file', :parameters => {:namevar => "$pup_tmp_filename", :content => "Sensitive.new('sekritD')"}, :pre_code => "$pup_tmp_filename = if $osfamily == 'windows' { '#{tmp_filename_win}' } else { '#{tmp_filename_else}' }",
+       :assertions => [{:refute_match => 'sekritD'}, {:assert_match => /#{tmp_environment}\.txt..content. #{file_redacted}/}]},
+
+  ]
+
+  sitepp_content = generate_manifest(test_resources)
+  assertion_code = generate_assertions(test_resources)
+
+  create_sitepp(master, tmp_environment, sitepp_content)
+
+  step "run agent in #{tmp_environment}, run all assertions" do
+    with_puppet_running_on(master,{}) do
+      agents.each do |agent|
+        on(agent, puppet("agent -t --debug --trace --show_diff --server #{master.hostname} --environment #{tmp_environment}"),
+           :accept_all_exit_codes => true) do |result|
+          assert(result.exit_code==2,'puppet agent run failed')
+          run_assertions(assertion_code, result)
+        end
+
+        if agent['platform'] =~ /fedora|centos|el|redhat|scientific|ubuntu|debian|cumulus/
+          step "assert no redacted data in syslog" do
+            #sigh.  gee, thanks for the help, beaker!
+            result = dump_puppet_log(agent)
+            raise if result.length < 3
+            string_io_index = 2
+            run_assertions(assertion_code, result[string_io_index].string)
+          end
+        end
+
+        # don't do this before the agent log scanning, above. it will skew the results
+        step "assert no redacted data in vardir" do
+          on(agent, "grep -R sekrit #{agent.puppet['vardir']}", :accept_all_exit_codes => true) do |result|
+            refute_match(/sekrit(1|2|3|6|7)/, result.stdout, 'found redacted data we should not have')
+            #TODO: if/when this is fixed, we should just be able to eval(assertion_code_ in this result block also!
+            expect_failure 'file resource contents will end up in the cached catalog en-clair' do
```

Comment:
This passes unexpectedly on solaris10-x64 using `1.6.2.133.g82d98be`, but not on Windows.

```
Warning: tests/language/sensitive_data_type.rb errored in 53.77 seconds
Test Case tests/language/sensitive_data_type.rb reported: #<RuntimeError: PuppetAcceptance::DSL::Helpers.with_puppet_running_on failed (check backtrace for location) because: An assertion was expected to fail, but passed. This is probably because a product bug was fixed, and "expect_failure()" needs to be removed from this test. Additional info: 'file resource contents will end up in the cached catalog en-clai
```